app/controllers/links_controller.rb

```diff
@@ -1,22 +1,24 @@
-class LinksController < ApplicationController 
-   skip_before_action :verify_authenticity_token
-   def create
-     @link = Link.new
-     @link.url = params[:url]
-     if @link.save
-        # loclhost/show?link=2
-       redirect_to root_path({link_id: @link.id})
-     else
+# frozen_string_literal: true
+
+class LinksController < ApplicationController
+  require 'valid_url.rb'
```

Comment:
You no need to require them. All directories inside of `app` folder autoload during Rails loading